pandas/io/pytables.py

```diff
@@ -3219,7 +3221,12 @@ def read(
         self.validate_read(columns, where)
         index = self.read_index("index", start=start, stop=stop)
         values = self.read_array("values", start=start, stop=stop)
-        return Series(values, index=index, name=self.name, copy=False)
+        result = Series(values, index=index, name=self.name, copy=False)
+        if result.dtype.kind == "O" and using_pyarrow_string_dtype():
```

Comment:
Not too familiar with this code, but do we need to check if results is a string array first if doing this?

Reply:
Yeah that makes sense